api/core/types.proto

```diff
@@ -10,38 +10,32 @@ message Interval {
     int64 seconds = 3;
 }
 
-enum FluxObjectKind {
-    KindGitRepository  = 0;
-    KindBucket         = 1;
-    KindHelmRepository = 2;
-    KindHelmChart      = 3;
-    KindKustomization  = 4;
-    KindHelmRelease    = 5;
-    KindCluster        = 6;
-    KindOCIRepository  = 7;
-    KindProvider       = 8;
-    KindAlert          = 9;
+enum Kind {
+    GitRepository  = 0;
+    Bucket         = 1;
+    HelmRepository = 2;
+    HelmChart      = 3;
+    Kustomization  = 4;
+    HelmRelease    = 5;
+    Cluster        = 6;
+    OCIRepository  = 7;
+    Provider       = 8;
+    Alert          = 9;
 };
 
 enum HelmRepositoryType {
     Default = 0;
     OCI     = 1;
 };
 
-message FluxObjectRef {
-    FluxObjectKind kind      = 1;
-    string         name      = 2;
-    string         namespace = 3;
-}
-
 message ObjectRef {
-    string         kind      = 1;
+    Kind           kind      = 1;
```

Comment:
:x: I missed this before - this changes the same "can't dynamically add more kinds and have the endpoint work" thing for listEvents that I was mentioning for listObjects and getObject

Reply:
Same thing here - every time we reference ObjectRef / ClusteredObjectRef in the backend this will have to be a string and not typed?